src/microbe_stage/organelle_components/MovementComponent.cs

```diff
@@ -30,6 +32,17 @@ public MovementComponent(float momentum)
 
     public void OnAttachToCell(PlacedOrganelle organelle)
     {
+        var configuration = organelle.Upgrades?.CustomUpgradeData;
+
+        if (configuration == null)
+        {
+            flagellumLength = 0;
```

Comment:
Does this result in flagellum being the length it is currently? Because for all upgrades when the data is missing I've made sure it results in the behaviour that happens if the upgrade didn't exist at all.

Reply:
Length for all related stuff is BASE + flagellumLength * modifier so if it's 0 stuff should be their base form

Reply:
That sounds good.